.github/workflows/production-heartbeat.yml

```diff
@@ -0,0 +1,73 @@
+name: production-heartbeat
+on:
+  schedule:
+    # Times are UTC
+    - cron: '45 1,5,9,13,17,21 * * *'
+jobs:
+  production-heartbeat:
+    strategy:
+      # By default, if any job in a matrix fails, all other jobs are immediately cancelled. This makes the jobs run to completion instead.
+      fail-fast: false
+      matrix:
+        os: [{vm: ubuntu-latest, exe: .sh}, {vm: windows-2019, exe: .cmd}]
+        node: ['lts/*']
+    runs-on: ${{ matrix.os.vm }}
+    steps:
+      # === Setup. We need to get the code, set up nodejs, and create the results directory. ===
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - run: mkdir smoke-test-results
+      # === Make three attempts to install sfdx through npm ===
+      - name: Install SFDX
+        id: sfdx_install
+        # Sleep for 5 seconds between attempts, since it's possible that problems are network-related.
+        run: npm install -g sfdx-cli || (sleep 5 && npm install -g sfdx-cli) || (sleep 5 && npm install -g sfdx-cli)
+
+      # === Make three attempts to install the scanner plugin ===
+      - name: Install Scanner Plugin
+        id: scanner_install
+        # Sleep for 5 seconds between attempts, since it's possible that problems are network-related.
+        run: sfdx plugins:install @salesforce/sfdx-scanner || (sleep 5 && sfdx plugins:install @salesforce/sfdx-scanner) || (sleep 5 && sfdx plugins:install @salesforce/sfdx-scanner)
+
+      # === Make three attempts to execute the smoke tests ===
+      - name: Run smoke tests
+        id: smoke_tests
+        # Sleep for 5 seconds between attempts, since it's possible that problems are network-related.
+        run: smoke-tests/smoke-test${{ matrix.os.exe }} sfdx || (sleep 5 && smoke-tests/smoke-test${{ matrix.os.exe }} sfdx) || (sleep 5 && smoke-tests/smoke-test${{ matrix.os.exe }} sfdx)
+
+      # === Upload the smoke-test-results folder as an artifact ===
+      - name: Upload smoke-test-results folder as artifact
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: smoke-test-results-${{ runner.os }}
+          path: smoke-test-results
+
+      # === Report any failures ===
+      - name: Report failures
+        if: ${{ failure() }}
+        shell: bash
+        env:
+          SFDX_STATUS: ${{ steps.sfdx_install.outcome == 'success' }}
```

Comment:
Could this output the `steps.sfdx_install.outcome` instead of converting to a boolean? That would provide more granularity on types of failures?

If we did this, we would want to change line 64 to be `status:` instead of `succeeded:`

Reply:
The granularity won't necessarily increase, because the possible states are "success", "failure", "cancelled", and "skipped". Steps that don't execute because a previous step failed will get the status "skipped", which feels like a misnomer to me but whatever.
Still, good point that the status can be a string now. I'll make that change.